packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -4079,16 +4079,20 @@ function flushRenderPhaseStrictModeWarningsInDEV() {
 function recursivelyTraverseAndDoubleInvokeEffectsInDEV(
   root: FiberRoot,
   parentFiber: Fiber,
+  treeFlags: Flags,
```

Comment:
1. The problem is that in this scenario only the root gets `PlacementDev` flag because [`reconcileChildren`](https://github.com/facebook/react/blob/ff6283340a10bb72ad0fb16ca027606a9ea1e67c/packages/react-reconciler/src/ReactFiberBeginWork.js#L337) routes to `mountChildFibers` for all of the descendant's fibers.
2. That in turn is a reconciler function created with `shouldTrackSideEffects === false`
3. So `.subtreeFlags` can't be relied upon because, well, those flags are not set on those descendant fibers so they are also not propagated onto their ancestors' `.subtreeFlags`

That led me to propagate the `treeFlags` through the callers' chain here